CHANGELOG.md

```diff
@@ -39,19 +39,19 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
   - Clarified that effective cost does not mix or "blend" costs across multiple charges.
 - `ListUnitPrice` column updates:
   - Column is conditional and only required when the provider publishes a price list that excludes discounts.
-  - Must not be null when `ChargeClass` is "Regular" and `ChargeCategory` is "Usage" or "Purchase".
+  - Must not be null when `ChargeClass` is not "Correction" and `ChargeCategory` is "Usage" or "Purchase".
```

Comment:
What about in-billing-period corrections? (This question applies to all of these `not "Correction"` changes. I think we want it to stay null if it's a correction within the billing period, right?) 

I feel like this requires us to have a value for in-period corrections so these statements can be correct. This feels like a blocker, right? This is the only reason I'm not signing off as-is.

Reply:
Good catch! We need to allow nulls for corrections that occur during the same billing period as well.!

It seems like we should provide the ability to identify/reference both scenarios, meaning we should provide two allowed values for corrections after all.

Reply:
would a correction create a new billing line for the previous charge or would the previous charge line item be updated?

Reply:
Do we think it's necessary to identify corrections that occur within the same billing period?

Reply:
@AWS-ZachErdman, based on our convos, different providers may handle this scenario differently. We don't think FOCUS needs to explicitly state how it should be handled today, but we do need to make sure that FOCUS is capable of representing the data however the provider emits charge rows. I may be missing a scenario, but we talked about:
1. Rewrite the original row (only 1 record post-correction)
2. Ledger-based amendments (old record + negation of the old record + all-inclusive new record)
3. Streaming data (old record + 1 or more delta records)

Reply:
Thanks @flanakin I understand your comment now. And yes, aligned with your correction scenarios.

I realize this is about the definition of ListUnitPrice. Should we just remove here (ListUnitPrice and also ContractUnitPrice definitions) the parts of the requirements that reference ChargeClass? Or is it adding something Im not realizing

Reply:
My personal view is corrections being separated out within billing periods makes things harder for billing data consumers (when providers restate the billing period). I've never cared to look for what was the first line a provider sent vs the next revision for the same line item - so my answer is based on that. Others might care to find differences - but that also means they're constantly having to aggregate to get the full picture.

For providers that send delta updates, it's unavoidable and therefore a users are trained to look for those deltas throughout the month. They can identify the original or subsequent updates based on when those charges came in.

Reply:
Naming aside, if we revert to  `Allow nulls: False` and the 3 allowed values - `Original`, `Current-cycle Correction`, and `Past-cycle Corrections`:

- When providers restate the billing period, you would only receive `Original` and `Past-cycle Corrections`.
- For providers that already send delta updates, you would receive charges with all 3 values.

If we specify all 3 allowed ChargeClass values, the normative for SkuId, SkuPriceId, PricingUnit, etc., is clear and straightforward. If not, we cannot use ChargeClass to identify when those columns MUST NOT be null.

Example: 
> It MUST NOT be null when ChargeClass is "Current-cycle Correction" or "Past-cycle Corrections," and ChargeCategory is "Usage" or "Purchase.

**Question:** Assuming we'll find suitable terms for all 3 allowed values :), are you strongly opposed to reverting to `Not-nullable` and having `Original` and two types of Corrections as allowed values?

Reply:
The team decided on a nullable column approach, where Correction is the sole allowed value